### DISCUSSION TITLE
Use case for paths aliases that don't start with special character

### DISCUSSION THREAD

**Discussion by thomasmattheussen** [vikejs/vike#1069] [Discussion]
I'm maintaining a multi-branded design system at the company I work at and we have some of our aliases set up like this:

```js
resolve: {
  alias: {
    '@mycompany/theme-whitelabel': `@mycompany/theme-${process.env.MH_THEME}`,
  },
},
```

Consumers of our design system write their imports like this `import x from '@mycompany/theme-whitelabel'` and based on their environment variables it gets aliased to the correct brand.

But recently, vite-plugin-ssr enforces your `resolve` aliases to be distinguishable from regular package imports. Where's this coming from exactly? Is there a workaround somehow that doesn't require another alias setup?

**Comment by brillout** [comment]
Yes, exactly, vite-plugin-ssr (and not only) needs to be able to distinguish between alias names and npm package names.

Can you rename your aliases to e.g. ``'#mycomany/theme-...`: `@mycompany/theme-${...}` ``?

**Reply by thomasmattheussen** [reply]
The company's code is rife with them, spread across different teams etc. Will be a pain :/

Out of curiosity, what do you mean by "and not only", will this be something that's coming across the entire "eco system", or..?

Also, is the `#` convention something coming from you guys, or is there a wider convention across the community?

**Reply by brillout** [reply]
> The company's code is rife with them, spread across different teams etc. Will be a pain :/

Correct me if I'm wrong, but AFAICT it's just a matter of applying some `sed` bash script which should work fairly quickly and reliably. (I understand that applying it across several teams is time consuming, but it's still fairly trivial.)

In general, if you break semantics then you're asking for trouble. And, by a matter of fact, VPS does need to disambiguate. (I stumbled upon another tool a while ago that also needs this but I don't remember which tool it was.)
 
> Also, is the `#` convention something coming from you guys, or is there a wider convention across the community?

https://vite-plugin-ssr.com/path-aliases#vite

Btw., since you seem to extensively use vite-plugin-ssr: would your company be up for [sponsoring](https://github.com/sponsors/brillout)?

**Reply by thomasmattheussen** [reply]
Maybe I've expressed myself somewhat confusingly. We already have this resolving mechanism going on for 4-5 years in different setups (webpack, vite, ...). It's not that we're using VPS accross the entire company, in fact we're only using it for a small part in our documentation website where we're pre-rendering a few component demos to static files. 

The reason why I was asking some background info, is that we'd have to introduce a new company wide resolving alias across each team's codebase "because otherwise our documentation demo's won't build anymore" 😅 

Because everything else in the node ecosystem doesn't seem to have this limitation, I was wondering to myself: where is this coming from? Will the eco system follow this because it's something Node introduced in a new version? Then I thought, I'll just ask 😄 

Anyway thanks for your replies!

(Regarding the sponsoring, our company (alas) doesn't have any procedures for this kind of thing at the moment.)

**Reply by brillout** [reply]
I understand where you come from and you'll probably disagree, but I'd argue it's a good thing for your company to move away from a brittle alias implementation.

I could dig into how vite-plugin-ssr could lift this limitation, but I usually do these kind of extra mile work only for sponsors (or if it turns out to be a widespread problem).

> (Regarding the sponsoring, our company (alas) doesn't have any procedures for this kind of thing at the moment.)

If your company reconsiders, then I'd be more than happy to dig into a solution that is easier for you to use.

**Reply by thomasmattheussen** [reply]
I'm not disagreeing per see, just looking for some background info to better my understanding 😊 
If this is the way to go, of course the company should indeed move away from this kind of implementation!

Again, thanks!
